### DISCUSSION TITLE
Failed! The new version of the composer can't be copied

### DISCUSSION THREAD

**Discussion by baldgzr** [atrocore/atrocore#189] [General]
Following through the install instructions on Ubuntu 18.04 

The following command:

> sudo git clone https://github.com/atrocore/skeleton-pim-no-demo.git . && php composer.phar self-update && php composer.phar update
> Cloning into '.'...
> remote: Enumerating objects: 32, done.
> remote: Counting objects: 100% (32/32), done.
> remote: Compressing objects: 100% (27/27), done.
> remote: Total 32 (delta 9), reused 12 (delta 3), pack-reused 0
> Unpacking objects: 100% (32/32), done.

Generates this error:

> Cannot create cache directory /home/user/.composer/cache/repo/https---repo.packagist.org/, or directory is not writable. Proceeding without cache
> Cannot create cache directory /home/user/.composer/cache/files/, or directory is not writable. Proceeding without cache
> Failed! The new version of the composer can't be copied. Try to download manually 'https://packagist.atrocore.com/composer/composer.phar'.

I'm confused, any ideas or suggestions are much appreciated.

Thank you.

**Comment by rratsun** [comment]
Hello, you get this error, because you don't have permissions to create file. Try to do this from root user.

**Comment by baldgzr** [comment]
All the commands were done as sudo ...

Should I do su?

**Comment by rratsun** [comment]
You added sudo only for git clone command. Please, add sudo for else. Example: sudo php composer.phar self-update

**Reply by baldgzr** [reply]
Thank you.

OK, That worked and the install has progressed up to the wizard, in the browser I am getting the error Error: Not Found (404).

What did I miss?

**Reply by rratsun** [reply]
Now you have wrong configuration for apache or nginx. Please, read [installation guide](https://github.com/atrocore/atrocore-docs/blob/master/en/administration/installation.md) and do all steps.

**Reply by baldgzr** [reply]
Quite possibly, I skipped the virtual host configuration, because the server is a VM and will be dedicated to AtroPIM. Access by IP address only. Could that be the issue?

**Reply by rratsun** [reply]
Sure

**Reply by baldgzr** [reply]
Tested that the apache2 config is working, so can load a file from the project directory OK!

However, the 404 Error is from within the code and in the browsers debug it suggests the following error:

```
Source map error: Error: JSON.parse: unexpected end of data at line 2 column 1 of the JSON data
Resource URL: http://192.168.2.109/client/lib/underscore-min.js
Source Map URL: underscore-min.map
```
Any suggestions on how to correct this error?

**Reply by rratsun** [reply]
Did you enable mod_rewrite? Please, show me your virtual host configuration.

**Reply by baldgzr** [reply]
Got it working now, thank you.
